test/server/cards/02_SHD/leaders/FennecShandHonoringTheDeal.spec.ts

```diff
@@ -0,0 +1,103 @@
+                expect(context.rebelPathfinder.damage).toBe(1);
+                expect(context.isbAgent.damage).toBe(2);
+            });
+        });
+
+        describe('Fennec Shand\'s deployed ability', function () {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        base: 'echo-base',
+                        leader: { card: 'fennec-shand#honoring-the-deal', deployed: true },
+                        hand: ['reinforcement-walker', 'rebel-pathfinder', 'alliance-xwing', 'modded-cohort'],
+                        resources: 8
+                    },
+                    player2: {
+                        groundArena: ['isb-agent', 'battlefield-marine'],
+                    }
+                });
+            });
+
+            it('should allow the player to play a unit with cost 4 or less, paying its cost, and give it ambush (multiple times)', function () {
+                const { context } = contextRef;
+
+                // use fennec ability to play a unit with ambush (should not exhaust fennec)
+                context.player1.clickCard(context.fennecShand);
+                context.player1.clickPrompt('Play a unit that costs 4 or less from your hand. Give it ambush for this phase');
+                expect(context.player1).toBeAbleToSelectExactly([context.rebelPathfinder, context.allianceXwing, context.moddedCohort]);
+
+                // play rebel pathfinder
+                context.player1.clickCard(context.rebelPathfinder);
+                expect(context.rebelPathfinder).toBeInLocation('ground arena');
+                expect(context.player1.countExhaustedResources()).toBe(2); // no extra cost
+                expect(context.player1).toHavePassAbilityPrompt('Ambush');
+
+                // ambush isb agent
+                context.player1.clickPrompt('Ambush');
+                context.player1.clickCard(context.isbAgent);
+                expect(context.fennecShand.exhausted).toBeFalse();
+                expect(context.rebelPathfinder.exhausted).toBeTrue();
+                expect(context.rebelPathfinder.damage).toBe(1);
+                expect(context.isbAgent.damage).toBe(2);
+
+                context.player2.passAction();
+
+                // attack with fennec (to exhaust her)
+                context.player1.clickCard(context.fennecShand);
+                context.player1.clickPrompt('Attack');
+                context.player1.clickCard(context.p2Base);
+
+                context.player2.passAction();
+
+                // use fennec ability to play a unit with ambush (even if fennec is exhausted)
+                context.player1.clickCard(context.fennecShand);
+                expect(context.player1).toBeAbleToSelectExactly([context.allianceXwing, context.moddedCohort]);
+                context.player1.clickCard(context.moddedCohort);
+
+                // play a unit who already has Ambush
```

Comment:
These are great! There's one additional test it would be really good to have. This is one of the few actions in the game with no cost, so it gives us a chance to test that our logic for determining whether an action is allowed works right.

Can you add one additional test (could be a separate `describe`) where she is deployed and the player has no cards in hand that are targetable with her ability? Then we need to confirm that when clicked on, the game doesn't give the player the option to use her ability at all. Normally an action ability is always available b/c the cost can change game state, but obviously this is a different situation.